src/SourceLocator/Type/DirectorySourceLocator.php

```diff
@@ -0,0 +1,82 @@
+<?php
+/**
+ * Created by PhpStorm.
+ * User: HP
+ * Date: 09/24/2016
+ * Time: 5:29 AM
+ */
+
+namespace BetterReflection\SourceLocator\Type;
+
+
+use BetterReflection\Identifier\Identifier;
+use BetterReflection\Identifier\IdentifierType;
+use BetterReflection\Reflector\Reflector;
+
+class DirectorySourceLocator implements SourceLocator
+{
+
+    /**
+     * @var AggregateSourceLocator
+     */
+    private $aggregatedSourceLocator;
+
+
+    /**
+     * DirectorySourceLocator constructor.
+     * @param $directory string|array directory to scan
```

Comment:
Please don't allow mixed types. You either want an `array` or a `string`, not both. If a single directory is to be passed in, that's up to the user.